From 7d699af24a21ca2b2db2352a668538ff87470e83 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mkbestas@lineageos.org>
Date: Thu, 21 Feb 2019 19:20:11 +0200
Subject: [PATCH 4/9] sepolicy: Correctly label display.qservice per SoC

* msm8952/msm8976/msm8992/msm8994 use it as vndservice
* apq8084/msm8226/msm8909/msm8916/msm8960/msm8974 use it as service

Change-Id: Ia600a777afa0ad164e31d52bcb3a644c52c22450
---
 apq8084/service.te          | 1 +
 apq8084/service_contexts    | 1 +
 common/service.te           | 1 -
 common/service_contexts     | 1 -
 msm8226/service.te          | 1 +
 msm8226/service_contexts    | 1 +
 msm8909/service.te          | 1 +
 msm8909/service_contexts    | 1 +
 msm8916/service.te          | 1 +
 msm8916/service_contexts    | 1 +
 msm8952/vndservice.te       | 1 +
 msm8952/vndservice_contexts | 1 +
 msm8960/service.te          | 1 +
 msm8960/service_contexts    | 1 +
 msm8974/service.te          | 1 +
 msm8974/service_contexts    | 1 +
 msm8976/vndservice.te       | 1 +
 msm8976/vndservice_contexts | 1 +
 msm8992/vndservice.te       | 1 +
 msm8992/vndservice_contexts | 1 +
 msm8994/vndservice.te       | 1 +
 msm8994/vndservice_contexts | 1 +
 22 files changed, 20 insertions(+), 2 deletions(-)
 create mode 100644 apq8084/service.te
 create mode 100644 apq8084/service_contexts
 create mode 100644 msm8226/service.te
 create mode 100644 msm8226/service_contexts
 create mode 100644 msm8916/service.te
 create mode 100644 msm8916/service_contexts
 create mode 100644 msm8952/vndservice.te
 create mode 100644 msm8952/vndservice_contexts
 create mode 100644 msm8960/service.te
 create mode 100644 msm8960/service_contexts
 create mode 100644 msm8974/service.te
 create mode 100644 msm8974/service_contexts
 create mode 100644 msm8976/vndservice.te
 create mode 100644 msm8976/vndservice_contexts
 create mode 100644 msm8992/vndservice.te
 create mode 100644 msm8992/vndservice_contexts
 create mode 100644 msm8994/vndservice.te
 create mode 100644 msm8994/vndservice_contexts

diff --git a/apq8084/service.te b/apq8084/service.te
new file mode 100644
index 0000000..8e2839d
--- /dev/null
+++ b/apq8084/service.te
@@ -0,0 +1 @@
+type qdisplay_service,            service_manager_type;
diff --git a/apq8084/service_contexts b/apq8084/service_contexts
new file mode 100644
index 0000000..b019356
--- /dev/null
+++ b/apq8084/service_contexts
@@ -0,0 +1 @@
+display.qservice                               u:object_r:qdisplay_service:s0
diff --git a/common/service.te b/common/service.te
index afc2b1f..14e9153 100644
--- a/common/service.te
+++ b/common/service.te
@@ -13,4 +13,3 @@ type dtseagleservice_service,     service_manager_type;
 type gba_auth_service,            service_manager_type;
 type mdtpdaemon_service,          service_manager_type;
 type qtitetherservice_service,    service_manager_type;
-type qdisplay_service,            service_manager_type;
diff --git a/common/service_contexts b/common/service_contexts
index aa7bac4..7cea549 100644
--- a/common/service_contexts
+++ b/common/service_contexts
@@ -45,4 +45,3 @@ com.qualcomm.qti.uceservice                    u:object_r:imsrcs_service:s0
 # DOLBY_START
 media.dolby_memoryservice                      u:object_r:audioserver_service:s0
 # DOLBY_END
-display.qservice                               u:object_r:qdisplay_service:s0
diff --git a/msm8226/service.te b/msm8226/service.te
new file mode 100644
index 0000000..8e2839d
--- /dev/null
+++ b/msm8226/service.te
@@ -0,0 +1 @@
+type qdisplay_service,            service_manager_type;
diff --git a/msm8226/service_contexts b/msm8226/service_contexts
new file mode 100644
index 0000000..b019356
--- /dev/null
+++ b/msm8226/service_contexts
@@ -0,0 +1 @@
+display.qservice                               u:object_r:qdisplay_service:s0
diff --git a/msm8909/service.te b/msm8909/service.te
index 7017041..e492acb 100644
--- a/msm8909/service.te
+++ b/msm8909/service.te
@@ -25,3 +25,4 @@
 # OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN
 # IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+type qdisplay_service,            service_manager_type;
diff --git a/msm8909/service_contexts b/msm8909/service_contexts
index 7017041..0d6cc02 100644
--- a/msm8909/service_contexts
+++ b/msm8909/service_contexts
@@ -25,3 +25,4 @@
 # OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN
 # IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+display.qservice                               u:object_r:qdisplay_service:s0
diff --git a/msm8916/service.te b/msm8916/service.te
new file mode 100644
index 0000000..8e2839d
--- /dev/null
+++ b/msm8916/service.te
@@ -0,0 +1 @@
+type qdisplay_service,            service_manager_type;
diff --git a/msm8916/service_contexts b/msm8916/service_contexts
new file mode 100644
index 0000000..b019356
--- /dev/null
+++ b/msm8916/service_contexts
@@ -0,0 +1 @@
+display.qservice                               u:object_r:qdisplay_service:s0
diff --git a/msm8952/vndservice.te b/msm8952/vndservice.te
new file mode 100644
index 0000000..b9cc327
--- /dev/null
+++ b/msm8952/vndservice.te
@@ -0,0 +1 @@
+type qdisplay_service,            vndservice_manager_type;
diff --git a/msm8952/vndservice_contexts b/msm8952/vndservice_contexts
new file mode 100644
index 0000000..b019356
--- /dev/null
+++ b/msm8952/vndservice_contexts
@@ -0,0 +1 @@
+display.qservice                               u:object_r:qdisplay_service:s0
diff --git a/msm8960/service.te b/msm8960/service.te
new file mode 100644
index 0000000..8e2839d
--- /dev/null
+++ b/msm8960/service.te
@@ -0,0 +1 @@
+type qdisplay_service,            service_manager_type;
diff --git a/msm8960/service_contexts b/msm8960/service_contexts
new file mode 100644
index 0000000..b019356
--- /dev/null
+++ b/msm8960/service_contexts
@@ -0,0 +1 @@
+display.qservice                               u:object_r:qdisplay_service:s0
diff --git a/msm8974/service.te b/msm8974/service.te
new file mode 100644
index 0000000..8e2839d
--- /dev/null
+++ b/msm8974/service.te
@@ -0,0 +1 @@
+type qdisplay_service,            service_manager_type;
diff --git a/msm8974/service_contexts b/msm8974/service_contexts
new file mode 100644
index 0000000..b019356
--- /dev/null
+++ b/msm8974/service_contexts
@@ -0,0 +1 @@
+display.qservice                               u:object_r:qdisplay_service:s0
diff --git a/msm8976/vndservice.te b/msm8976/vndservice.te
new file mode 100644
index 0000000..b9cc327
--- /dev/null
+++ b/msm8976/vndservice.te
@@ -0,0 +1 @@
+type qdisplay_service,            vndservice_manager_type;
diff --git a/msm8976/vndservice_contexts b/msm8976/vndservice_contexts
new file mode 100644
index 0000000..b019356
--- /dev/null
+++ b/msm8976/vndservice_contexts
@@ -0,0 +1 @@
+display.qservice                               u:object_r:qdisplay_service:s0
diff --git a/msm8992/vndservice.te b/msm8992/vndservice.te
new file mode 100644
index 0000000..b9cc327
--- /dev/null
+++ b/msm8992/vndservice.te
@@ -0,0 +1 @@
+type qdisplay_service,            vndservice_manager_type;
diff --git a/msm8992/vndservice_contexts b/msm8992/vndservice_contexts
new file mode 100644
index 0000000..b019356
--- /dev/null
+++ b/msm8992/vndservice_contexts
@@ -0,0 +1 @@
+display.qservice                               u:object_r:qdisplay_service:s0
diff --git a/msm8994/vndservice.te b/msm8994/vndservice.te
new file mode 100644
index 0000000..b9cc327
--- /dev/null
+++ b/msm8994/vndservice.te
@@ -0,0 +1 @@
+type qdisplay_service,            vndservice_manager_type;
diff --git a/msm8994/vndservice_contexts b/msm8994/vndservice_contexts
new file mode 100644
index 0000000..b019356
--- /dev/null
+++ b/msm8994/vndservice_contexts
@@ -0,0 +1 @@
+display.qservice                               u:object_r:qdisplay_service:s0
-- 
2.17.1

